lib/rust/ensogl/component/flame-graph/src/lib.rs

```diff
@@ -0,0 +1,298 @@
+//! An EnsoGL implementation of a basic flame graph.
+
+#![warn(missing_copy_implementations)]
+#![warn(missing_debug_implementations)]
+#![warn(missing_docs)]
+#![warn(trivial_casts)]
+#![warn(trivial_numeric_casts)]
+#![warn(unsafe_code)]
+#![warn(unused_import_braces)]
+#![warn(unused_qualifications)]
+
+
+use ensogl_core::prelude::*;
+
+use enso_frp as frp;
+use enso_profiler as profiler;
+use ensogl_core::application;
+use ensogl_core::application::shortcut;
+use ensogl_core::application::Application;
+use ensogl_core::data::color;
+use ensogl_core::display;
+use ensogl_core::display::scene::Layer;
+use ensogl_core::display::shape::*;
+use ensogl_core::Animation;
+use ensogl_text as text;
+
+
+
+// =======================
+// === Block Component ===
+// =======================
+
+/// A single block that makes up a flame graph.
+mod block {
+    use super::*;
+
+
+    // === Layout Constants ===
+
+    const TEXT_OFFSET_X: f32 = 4.0;
+    const TEXT_OFFSET_Y: f32 = -2.0;
+
+
+    // === Shapes Definitions ===
+
+    mod background {
+        use super::*;
+        ensogl_core::define_shape_system! {
+            (style:Style) {
+                let width  : Var<Pixels> = "input_size.x".into();
+                let height : Var<Pixels> = "input_size.y".into();
+                let zoom                 = Var::<f32>::from("1.0/zoom()");
+                let base_color           = style.get_color("flame_graph_color");
+
+                let shape = Rect((&width,&height));
+
+                let border_width: Var<Pixels> = (zoom * 2.0).into();
+
+                let right = Rect((&border_width,&height));
+                let right = right.translate_x(&width/2.0);
+
+                let left = Rect((&border_width,&height));
+                let left = left.translate_x(-&width/2.0);
+
+                let shape = shape - left;
+                let shape = shape - right;
+                let shape = shape.fill(base_color);
+
+                (shape).into()
+            }
+        }
+    }
+
+
+    // === FRP ===
+
+    ensogl_core::define_endpoints! {
+        Input {
+            set_content(String),
+            set_position(Vector2),
+            set_size(Vector2)
+        }
+        Output {}
+    }
+
+    // === Model ===
+
+    #[derive(Clone, Debug)]
+    struct Model {
+        app:            Application,
+        background:     background::View,
+        label:          text::Area,
+        display_object: display::object::Instance,
+    }
+
+    impl Model {
+        fn new(app: Application) -> Self {
+            let scene = app.display.scene();
+            let logger = Logger::new("FlameGraphBlock");
+            let display_object = display::object::Instance::new(&logger);
+
+            let label = app.new_view::<text::Area>();
+
+            let background = background::View::new(&logger);
+
+            display_object.add_child(&background);
+            display_object.add_child(&label);
+
+            let app = app.clone_ref();
```

Comment:
what is the reason for all these newlines in the middle of this function? I understand that you want to separate visual creation of things and adding them as children, but I think we should use fewer of such visual indicators (or none at all). What do you think?